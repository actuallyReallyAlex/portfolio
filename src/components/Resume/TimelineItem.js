import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Paragraph, Text } from 'grommet'
import * as icons from 'grommet-icons'

const TimelineItem = ({
  endDate,
  icon,
  organization,
  present,
  startDate,
  summary,
  title
}) => {
  const Icon = icons[icon]

  return (
    <Box
      align="start"
      direction="row"
      fill="horizontal"
      gap="xlarge"
      justify="center"
    >
      <Box style={{ textAlign: 'right' }} width="30%">
        <Heading level="3">{title}</Heading>
        <Text>
          {startDate} - {present ? 'Present' : endDate}
        </Text>
      </Box>
      <Box align="center" pad="medium" width="10%">
        <Icon />
      </Box>
      <Box>
        <Heading level="4" margin="none">
          {organization}
        </Heading>
        <Box
          background="neutral-3"
          height="3px"
          margin={{ top: 'small', bottom: 'medium' }}
          round="small"
          width="60%"
        />
        <Paragraph>{summary}</Paragraph>
      </Box>
    </Box>
  )
}

TimelineItem.propTypes = {
  endDate: PropTypes.string,
  icon: PropTypes.string.isRequired,
  organization: PropTypes.string.isRequired,
  present: PropTypes.bool,
  startDate: PropTypes.string,
  summary: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default TimelineItem
