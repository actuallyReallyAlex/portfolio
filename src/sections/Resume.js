import React from 'react'
import { Section } from 'grommet'
import SectionHeader from '../components/SectionHeader'
import Experience from '../components/Experience'

const Resume = () => {
  return (
    <Section id="resume">
      <SectionHeader
        title="Résumé"
        subtitle="More of my credentials."
        summary="Lorem ipsum Exercitation culpa qui dolor consequat exercitation
            fugiat laborum ex ea eiusmod ad do aliqua occaecat nisi ad irure
            sunt id pariatur Duis laboris amet exercitation veniam labore
            consectetur ea id quis eiusmod."
      />
      <Experience />
    </Section>
  )
}

export default Resume
