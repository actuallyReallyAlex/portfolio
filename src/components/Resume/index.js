import React from 'react'
import { Box, Button, Paragraph } from 'grommet'
import { Download } from 'grommet-icons'
import SectionHeading from '../SectionHeading'
import { education, experience, resume } from '../../config.json'
import TimelineItem from './TimelineItem'

const Resume = ({ navRef }) => {
  return (
    <Box
      align="center"
      fill="horizontal"
      gap="medium"
      id="Résumé"
      justify="center"
      pad="large"
      ref={navRef}
    >
      <SectionHeading heading="RÉSUMÉ" subHeading="More of my credentials" />
      <Paragraph>{resume.copy}</Paragraph>
      <Button href={resume.url} icon={<Download />} label="Résumé" primary />
      <SectionHeading subHeading="Work Experience" />
      <Box fill="horizontal" gap="small" margin={{ vertical: 'medium' }}>
        {experience.map((props, i) => (
          <TimelineItem key={i} {...props} />
        ))}
      </Box>
      <SectionHeading subHeading="Education" />
      <Box fill="horizontal" gap="small" margin={{ vertical: 'medium' }}>
        {education.map((props, i) => (
          <TimelineItem key={i} {...props} />
        ))}
      </Box>
    </Box>
  )
}

export default Resume
