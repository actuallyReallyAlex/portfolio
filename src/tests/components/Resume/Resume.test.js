import React from 'react'
import Resume from '../../../components/Resume'
import { shallow } from 'enzyme'

const defaultProps = {
  navRef: jest.fn(),
  size: 'medium'
}

describe('<Resume />', () => {
  test('Should render the <Resume />.', () => {
    const component = shallow(<Resume {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <Resume /> when viewed from a small device.', () => {
    const component = shallow(<Resume {...defaultProps} size="small" />)
    expect(component).toMatchSnapshot()
  })
})
