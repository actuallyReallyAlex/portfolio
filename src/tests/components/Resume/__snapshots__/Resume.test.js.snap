// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resume /> Should render the <Resume /> when viewed from a small device. 1`] = `ShallowWrapper {}`;

exports[`<Resume /> Should render the <Resume />. 1`] = `ShallowWrapper {}`;
